cvat-ui error when open a job with issues

### Actions before raising this issue

- [X] I searched the existing issues and did not find anything similar.
- [X] I read/searched [the docs](https://opencv.github.io/cvat/docs/)

### Steps to Reproduce

1. create a issue in a job
2. open the job
3. go to the Jobs
4. click on the job I entered earlier

### Expected Behavior

_No response_

### Possible Solution

The error may be caused by HiddenIssueLabel.

### Context

error:
react-dom.production.min.js:209 Error: Minified React error #200; visit https://reactjs.org/docs/error-decoder.html?invariant=200 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at Object.Zl [as createPortal] (react-dom.production.min.js:287:319)
    at $Te (hidden-issue-label.tsx:65:12)
    at qi (react-dom.production.min.js:153:146)
    at ys (react-dom.production.min.js:261:496)
    at fl (react-dom.production.min.js:246:265)
    at dl (react-dom.production.min.js:246:194)
    at rl (react-dom.production.min.js:239:172)
    at react-dom.production.min.js:123:115
    at t.unstable_runWithPriority (scheduler.production.min.js:19:467)
    at Va (react-dom.production.min.js:122:325)

### Environment

_No response_

--- a/changelog.d/20240507_103107_boris_fixed_exception.md
+++ b/changelog.d/20240507_103107_boris_fixed_exception.md
@@ -1,0 +1,4 @@
+### Fixed
+
+- Cannot read property 'annotations' of null when uploading annotations into a job
+  (<https://github.com/cvat-ai/cvat/pull/7857>)

--- a/cvat-ui/package.json
+++ b/cvat-ui/package.json
@@ -1,6 +1,6 @@
 {
   "name": "cvat-ui",
-  "version": "1.63.9",
+  "version": "1.63.10",
   "description": "CVAT single-page application",
   "main": "src/index.tsx",
   "scripts": {

--- a/cvat-ui/src/actions/annotation-actions.ts
+++ b/cvat-ui/src/actions/annotation-actions.ts
@@ -869,10 +869,17 @@ export function resetCanvas(): AnyAction {
 }
 
 export function closeJob(): ThunkAction {
-    return async (dispatch: ActionCreator<Dispatch>): Promise<void> => {
+    return async (dispatch: ActionCreator<Dispatch>, getState): Promise<void> => {
+        const state = getState();
+        const { instance: canvasInstance } = state.annotation.canvas;
         const { jobInstance } = receiveAnnotationsParameters();
+
         if (jobInstance) {
             await jobInstance.close();
+        }
+
+        if (canvasInstance) {
+            canvasInstance.destroy();
         }
 
         dispatch({

--- a/cvat-ui/src/actions/import-actions.ts
+++ b/cvat-ui/src/actions/import-actions.ts
@@ -5,7 +5,9 @@
 
 import { createAction, ActionUnion, ThunkAction } from 'utils/redux';
 import { CombinedState } from 'reducers';
-import { getCore, Storage } from 'cvat-core-wrapper';
+import {
+    getCore, Storage, Job, Task, Project,
+} from 'cvat-core-wrapper';
 import { EventScope } from 'cvat-logger';
 import { getProjectsAsync } from './projects-actions';
 import { AnnotationActionTypes, fetchAnnotationsAsync } from './annotation-actions';
@@ -36,10 +38,10 @@ export const importActions = {
     importDataset: (instance: any, format: string) => (
         createAction(ImportActionTypes.IMPORT_DATASET, { instance, format })
     ),
-    importDatasetSuccess: (instance: any, resource: 'dataset' | 'annotation') => (
+    importDatasetSuccess: (instance: Job | Task | Project, resource: 'dataset' | 'annotation') => (
         createAction(ImportActionTypes.IMPORT_DATASET_SUCCESS, { instance, resource })
     ),
-    importDatasetFailed: (instance: any, resource: 'dataset' | 'annotation', error: any) => (
+    importDatasetFailed: (instance: Job | Task | Project, resource: 'dataset' | 'annotation', error: any) => (
         createAction(ImportActionTypes.IMPORT_DATASET_FAILED, {
             instance,
             resource,
@@ -112,22 +114,18 @@ export const importDatasetAsync = (
                 await instance.logger.log(EventScope.uploadAnnotations);
                 await instance.annotations.clear(true);
                 await instance.actions.clear();
-                const history = await instance.actions.get();
 
                 // first set empty objects list
                 // to escape some problems in canvas when shape with the same
                 // clientID has different type (polygon, rectangle) for example
-                dispatch({
-                    type: AnnotationActionTypes.UPLOAD_JOB_ANNOTATIONS_SUCCESS,
-                    payload: {
-                        states: [],
-                        history,
-                    },
-                });
+                dispatch({ type: AnnotationActionTypes.UPLOAD_JOB_ANNOTATIONS_SUCCESS });
 
-                setTimeout(() => {
-                    dispatch(fetchAnnotationsAsync());
-                });
+                const relevantInstance = getState().annotation.job.instance;
+                if (relevantInstance && relevantInstance.id === instance.id) {
+                    setTimeout(() => {
+                        dispatch(fetchAnnotationsAsync());
+                    });
+                }
             }
         } catch (error) {
             dispatch(importActions.importDatasetFailed(instance, resource, error));

--- a/cvat-ui/src/components/annotation-page/annotation-page.tsx
+++ b/cvat-ui/src/components/annotation-page/annotation-page.tsx
@@ -4,7 +4,6 @@
 // SPDX-License-Identifier: MIT
 
 import React, { useEffect } from 'react';
-import { useHistory } from 'react-router';
 import Layout from 'antd/lib/layout';
 import Result from 'antd/lib/result';
 import Spin from 'antd/lib/spin';
@@ -44,7 +43,6 @@ export default function AnnotationPageComponent(props: Props): JSX.Element {
     const prevJob = usePrevious(job);
     const prevFetching = usePrevious(fetching);
 
-    const history = useHistory();
     useEffect(() => {
         saveLogs();
         const root = window.document.getElementById('root');
@@ -54,12 +52,10 @@ export default function AnnotationPageComponent(props: Props): JSX.Element {
 
         return () => {
             saveLogs();
+            closeJob();
+
             if (root) {
                 root.style.minHeight = '';
-            }
-
-            if (!history.location.pathname.includes('/jobs')) {
-                closeJob();
             }
         };
     }, []);

--- a/cvat-ui/src/reducers/annotation-reducer.ts
+++ b/cvat-ui/src/reducers/annotation-reducer.ts
@@ -779,15 +779,16 @@ export default (state = defaultState, action: AnyAction): AnnotationState => {
             };
         }
         case AnnotationActionTypes.UPLOAD_JOB_ANNOTATIONS_SUCCESS: {
-            const { states, history } = action.payload;
-
             return {
                 ...state,
                 annotations: {
                     ...state.annotations,
-                    history,
-                    states,
+                    history: { undo: [], redo: [] },
+                    states: [],
                     activatedStateID: null,
+                    activatedElementID: null,
+                    activatedAttributeID: null,
+                    highlightedConflict: null,
                     collapsed: {},
                 },
             };
@@ -1083,10 +1084,7 @@ export default (state = defaultState, action: AnyAction): AnnotationState => {
         }
         case AnnotationActionTypes.CLOSE_JOB:
         case AuthActionTypes.LOGOUT_SUCCESS: {
-            if (state.canvas.instance) {
-                state.canvas.instance.destroy();
-            }
-            return { ...defaultState };
+            return defaultState;
         }
         default: {
             return state;

--- a/cvat-ui/src/reducers/notifications-reducer.ts
+++ b/cvat-ui/src/reducers/notifications-reducer.ts
@@ -5,7 +5,7 @@
 
 import { AnyAction } from 'redux';
 
-import { ServerError } from 'cvat-core-wrapper';
+import { Project, ServerError, Task } from 'cvat-core-wrapper';
 import { AuthActionTypes } from 'actions/auth-actions';
 import { FormatsActionTypes } from 'actions/formats-actions';
 import { ModelsActionTypes } from 'actions/models-actions';
@@ -567,10 +567,17 @@ export default function (state = defaultState, action: AnyAction): Notifications
         }
         case ImportActionTypes.IMPORT_DATASET_SUCCESS: {
             const { instance, resource } = action.payload;
-            const message = resource === 'annotation' ?
-                'Annotations have been loaded to the ' +
-                `[task ${instance.taskId || instance.id}](/tasks/${instance.taskId || instance.id}) ` :
-                `Dataset was imported to the [project ${instance.id}](/projects/${instance.id})`;
+            let message = resource === 'annotation' ?
+                'Annotations have been loaded to the ' :
+                'Dataset was imported to the ';
+            if (instance instanceof Project) {
+                message += `[Project ${instance.id}](/projects/${instance.id})`;
+            } else if (instance instanceof Task) {
+                message += `[Task ${instance.id}](/tasks/${instance.id})`;
+            } else {
+                message += `[Job ${instance.id}](/jobs/${instance.id})`;
+            }
+
             return {
                 ...state,
                 messages: {

--- a/cvat-ui/src/reducers/review-reducer.ts
+++ b/cvat-ui/src/reducers/review-reducer.ts
@@ -197,7 +197,7 @@ export default function (state: ReviewState = defaultState, action: any): Review
         }
         case AnnotationActionTypes.CLOSE_JOB:
         case AuthActionTypes.LOGOUT_SUCCESS: {
-            return { ...defaultState };
+            return defaultState;
         }
         default:
             return state;

--- a/cvat-ui/src/reducers/tasks-reducer.ts
+++ b/cvat-ui/src/reducers/tasks-reducer.ts
@@ -1,5 +1,5 @@
 // Copyright (C) 2020-2022 Intel Corporation
-// Copyright (C) 2022-2023 CVAT.ai Corporation
+// Copyright (C) 2022-2024 CVAT.ai Corporation
 //
 // SPDX-License-Identifier: MIT
 
@@ -8,7 +8,6 @@ import { BoundariesActionTypes } from 'actions/boundaries-actions';
 import { TasksActionTypes } from 'actions/tasks-actions';
 import { AuthActionTypes } from 'actions/auth-actions';
 
-import { AnnotationActionTypes } from 'actions/annotation-actions';
 import { TasksState } from '.';
 
 const defaultState: TasksState = {
@@ -134,11 +133,6 @@ export default (state: TasksState = defaultState, action: AnyAction): TasksState
                     modalVisible: action.payload.visible,
                     taskId: action.payload.taskId,
                 },
-            };
-        }
-        case AnnotationActionTypes.CLOSE_JOB: {
-            return {
-                ...state,
             };
         }
         case BoundariesActionTypes.RESET_AFTER_ERROR:

